lib/commands/general.js

```diff
@@ -71,13 +71,27 @@ commands.background = async function (duration) {
 
 /*
  * Simulate Touch ID with either valid (match === true) or invalid (match === false)
- * fingerprint
+ * fingerprint (Simulator only)
  */
 commands.touchId = async function (match = true) {
-  let params = {
-    match
-  };
-  return await this.proxyCommand('/wda/touch_id', 'POST', params);
+  if (this.isSimulator()) {
+    return await this.proxyCommand('/wda/touch_id', 'POST', {match});
+  } else {
+    throw new errors.UnknownError('Touch ID simulation not supported on real devices');
+  }
+};
+
+/*
+ * Toggle enrollment of touchId (Simuulator only)
+ */
+commands.toggleEnrollTouchId = async function () {
+  if (!this.opts.allowTouchIdEnroll) {
```

Comment:
Same thing here. Also I'm not a big fan of the mixed conditional, if only because it makes the code harder to reason about. We have two errors conditions, and then the real deal.
```js
if (!this.opts.allowTouchIdEnroll) {
  throw new errors.UnknownError(`Must set desired capability 'allowTouchIdEnroll = true' to enroll touchId`);
}
if (!this.isSimulator()) {
  throw new errors.UnknownError('Touch ID simulation not supported on real devices');
}
await this.opts.device.enrollTouchID();
```